src/server/pkg/storage/fileset/option.go

```diff
@@ -65,10 +65,17 @@ type WriterOption func(w *Writer)
 // WithNoUpload sets the writer to no upload (will not upload chunks).
```

Comment:
I think it might be better to just nest the `WithIndexCallback` option here. Also, `f` -> `cb` if you would.

Reply:
What do you think about removing the callback from `WithNoUpload` and have the 2 options be totally orthogonal?

Reply:
I considered that approach, but I cannot come up with a use case for `NoUpload` without the index callback. As is, there would not be a way to collect or store any information with `NoUpload` without the index callback, which does not seem useful to me. I don't have a strong opinion either way, but I think having `NoUpload` include an index callback makes sense.